cryptoki/src/types/mechanism/mod.rs

```diff
@@ -280,3 +281,44 @@ impl TryFrom<psa_crypto::types::algorithm::Algorithm> for Mechanism {
         }
     }
 }
+
+/// Contains information about a mechanism
+#[derive(Debug, Clone, Copy, Default)]
+pub struct MechanismInfo {
+    val: CK_MECHANISM_INFO,
+}
+
+impl MechanismInfo {
+    pub(crate) fn new(val: CK_MECHANISM_INFO) -> Self {
+        Self { val }
+    }
+
+    /// Returns the minimum key size for this mechanism.
+    pub fn get_min_key_size(&self) -> u64 {
+        self.val.ulMinKeySize
+    }
+
+    /// Returns the maximum key size for this mechanism.
+    pub fn get_max_key_size(&self) -> u64 {
+        self.val.ulMaxKeySize
+    }
```

Comment:
I am unsure how to fix this, I'm currently contemplating just doing an `.into()`, which would convert `u32` to `u64` when necessary.

Reply:
These seem to be defined as `unsigned long` so they'd be platform-dependent. We have the [`Ulong`](https://docs.rs/cryptoki/0.1.1/cryptoki/types/struct.Ulong.html) type to cover it, though it's just a wrapper around the FFI type. But it might be best to offload the type conversion to the crate user.

Reply:
Done.